github/repos_test.go

```diff
@@ -656,14 +653,35 @@ func TestRepositoriesService_GetRequiredStatusChecks(t *testing.T) {
 	}
 }
 
+func TestRepositoriesService_GetRequiredStatusChecks_unprotectedBranch(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/branches/b/protection/required_status_checks", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeProtectedBranchesPreview)
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprint(w, `{
+  "message": "Not Found",
```

Comment:
In [my testing](https://github.com/google/go-github/issues/625#issuecomment-298063138), the message one gets when a branch is unprotected is "Branch not protected". It's "Not Found" only when the query is rejected because of insufficient permissions to the repo.

I think it's be good idea to display an accurate error message here, so that people in the future don't get confused.